validator/coordinator/src/main/java/tech/pegasys/artemis/validator/coordinator/ValidatorCoordinator.java

```diff
@@ -279,135 +277,83 @@ private void produceAttestations(
     this.eventBus.post(attestation);
   }
 
-  /**
-   * Gets the epoch signature used for RANDAO from the Validator Client using gRPC
-   *
-   * @param state
-   * @param proposer
-   * @return
-   * @see
-   *     <a>https://github.com/ethereum/eth2.0-specs/blob/v0.8.1/specs/validator/0_beacon-chain-validator.md#randao-reveal</a>
-   */
-  // TODO: since this is very similar to a spec function now, move it to a util file and
-  // abstract away the gRPC details
-  public BLSSignature get_epoch_signature(BeaconState state, BLSPublicKey proposer) {
-    UnsignedLong slot = state.getSlot().plus(UnsignedLong.ONE);
-    UnsignedLong epoch = BeaconStateUtil.compute_epoch_at_slot(slot);
-
-    Bytes32 messageHash =
-        HashTreeUtil.hash_tree_root(SSZTypes.BASIC, SSZ.encodeUInt64(epoch.longValue()));
-    Bytes domain = get_domain(state, Constants.DOMAIN_RANDAO, epoch);
-    return getSignature(messageHash, domain, proposer);
-  }
-
-  /**
-   * Gets the block signature from the Validator Client using gRPC
-   *
-   * @param state
-   * @param block
-   * @param proposer
-   * @return
-   * @see
-   *     <a>https://github.com/ethereum/eth2.0-specs/blob/v0.8.1/specs/validator/0_beacon-chain-validator.md#signature</a>
-   */
-  private BLSSignature getBlockSignature(
-      BeaconState state, BeaconBlock block, BLSPublicKey proposer) {
-    Bytes domain =
-        get_domain(
-            state,
-            Constants.DOMAIN_BEACON_PROPOSER,
-            BeaconStateUtil.compute_epoch_at_slot(block.getSlot()));
+  private void createBlockIfNecessary(
+      BeaconStateWithCache previousState, BeaconBlock previousBlock) {
+    // TODO - we shouldn't assume that we're just incrementing the slot by 1
+    final UnsignedLong newSlot = previousState.getSlot().plus(UnsignedLong.ONE);
```

Comment:
This was exactly what I was trying to incorporate into my current PR, but then decided not to:
https://pegasys1.atlassian.net/browse/BC-147